src/cdk-experimental/scrolling/virtual-scroll.md

```diff
@@ -0,0 +1,143 @@
+* `odd` - Whether the `index` is odd.
+
+All of these apply to the index of the item in the data source, not the index in the rendered
+portion of the data.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items;
+                       let index = index;
+                       let count = count;
+                       let first = first;
+                       let last = last;
+                       let even = even;
+                       let odd = odd">
+    {{item}} - {{index}} - {{count}} - {{first}} - {{last}} - {{even}} - {{odd}}
+  </div>
+</cdk-virtual-scroll-viewport>
+```
+
+A `trackBy` function can be specified and works the same as the `*ngFor` `trackBy`. The `index`
+passed to the tracking function will be the index in the data source, not the index in the rendered
+portion.
+
+#### View recycling
+In order to improve performance `*cdkVirtualFor` saves the views it creates in a cache when they are
+no longer needed. This way the next time a new view is needed one can be recycled from the cache
+rather than being created from scratch. The size of this cache can be adjusted using the
+`templateCacheSize` input. Caching can be disabled by setting it to `0`.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  <div *cdkVirtualFor="let item of items; templateCacheSize: 0">{{item}}</div>
+</cdk-virtual-scroll-viewport>
+```
+
+#### Specifying items with an Observable or DataSource
+`*cdkVirtualFor` is set up to accept data from an `Array`, `Observable`, or `DataSource`. The
+`DataSource` for the virtual scroll is the same one used by the table and tree components. A
+`DataSource` is simply an abstract class that has two methods: `connect` and `disconnect`. The
+`connect` method will be called by the virtual scroll viewport to receive a stream that emits the
+data array that should be rendered. The viewport will call `disconnect` when the viewport is
+destroyed, which may be the right time to clean up any subscriptions that were registered during the
+connect process.
+
+### Scrolling over fixed size items
+If you're scrolling over a list of items that are all the same fixed size, you can use the
+`FixedSizeVirtualScrollStrategy`. This can be easily added to your viewport using the `itemSize`
+directive.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50">
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+The fixed size strategy also supports setting the buffer size, i.e. the number of items rendered
+beyond the edge of the viewport. This can be adjusted by setting the `bufferSize` input.
+
+```html
+<cdk-virtual-scroll-viewport itemSize="50" bufferSize="1">
+  ...
+</cdk-virtual-scroll-viewport>
+```
+
+**Note: The fixed size strategy will likely be changed to allow specifying a separate
+`minBufferPx` and `addBufferPx` like the autosize strategy**
+
+The fixed size viewport strategy is less flexible than the autosize strategy because it requires all
```

Comment:
I would just explain this once near the beginning of the doc; it's currently mentioned a few times throughout